src/metal/metal-device.cpp

```
@@ -77,6 +76,25 @@ Result DeviceImpl::initialize(const DeviceDesc& desc, BackendImpl* backend)
+        return SLANG_FAIL;
+    }
+
+    // R2: Create residency set (mandatory — all allocations are registered here).
+    {
+        NS::Error* error = nullptr;
+        auto rsDesc = NS::TransferPtr(MTL::ResidencySetDescriptor::alloc()->init());
+        m_residencySet = NS::TransferPtr(m_device->newResidencySet(rsDesc.get(), &error));
+        if (!m_residencySet)
+        {
+            return SLANG_FAIL;
+        }
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)